pybamm/citations.py

```diff
@@ -150,23 +150,22 @@ def _parse_citation(self, key):
         key: str
             A BibTeX formatted citation
         """
-        PybtexError = have_optional_dependency("pybtex.scanner", "PybtexError")
-        parse_string = have_optional_dependency("pybtex.database", "parse_string")
+
+        # Parse string as a bibtex citation, and check that a citation was found
         try:
-            # Parse string as a bibtex citation, and check that a citation was found
-            bib_data = parse_string(key, bib_format="bibtex")
+            parse_string = have_optional_dependency("bibtexparser", "parse_string")
+            bib_data = parse_string(key)
             if not bib_data.entries:
-                raise PybtexError("no entries found")
+                raise Exception("no entries found")
 
             # Add and register all citations
-            for key, entry in bib_data.entries.items():
+            for entry in bib_data.entries:
                 # Add to _all_citations dictionary
-                self._add_citation(key, entry)
+                self._add_citation(entry.key, entry)
                 # Add to _papers_to_cite set
-                self._papers_to_cite.add(key)
+                self._papers_to_cite.add(entry.key)
                 return
-        except PybtexError:
```

Comment:
Does `bibtexparser` also have some attribute corresponding to `PybtexError`?

Reply:
No it doesn't. The only thing close to exception handling in bibtexparser is [this](https://bibtexparser.readthedocs.io/en/main/quickstart.html#:~:text=Step%202%3A%20Error%20Checking%C2%B6).